apis/actions.github.com/v1alpha1/autoscalingrunnerset_types.go

```diff
@@ -80,6 +85,124 @@ type ProxyConfig struct {
 
 	// +optional
 	HTTPS *ProxyServerConfig `json:"https,omitempty"`
+
+	// +optional
+	NoProxy []string `json:"noProxy,omitempty"`
+}
+
+func (c *ProxyConfig) ProxyFunc(secretFetcher func(string) (*corev1.Secret, error)) (func(*http.Request) (*url.URL, error), error) {
+	config := &httpproxy.Config{
+		NoProxy: strings.Join(c.NoProxy, ","),
+	}
+
+	if c.HTTP != nil {
+		u, err := url.Parse(c.HTTP.Url)
+		if err != nil {
+			return nil, err
```

Comment:
do you mind wrapping all errors to provide extra info, ex: `fmt.Errorf("failed to parse http proxy url: %v", err)`?